storage/engine/rocksdb.go

```diff
@@ -511,6 +511,9 @@ func (r *rocksDBIterator) Seek(key MVCCKey) {
 		// to access start[0] in an explicit seek.
 		r.setState(C.DBIterSeekToFirst(r.iter))
 	} else {
+		if r.valid && key.Equal(r.unsafeKey()) {
+			return
```

Comment:
Might add a comment about what is being done here: `We can avoid seeking if we're already at the key we seek.`